### PR TITLE
[BotBuilder-AI] Added documentation to luisClient.ts

### PR DESCRIPTION
**Pull request by Konigstiger**
## Description
The class `LuisClient` had missing comments. We received feedback to add them.

## Specific Changes
Add documentation to:
- enum `LuisApikeys`
- class `LuisClient`
- field `_basePath`

## Testing
The next image shows the result of the local tests.
image

### CODE REVIEW

File: libraries/botbuilder-ai/src/luis-client/luisClient.ts [hunk 1]
```diff
@@ -51,11 +51,23 @@ export interface PredictionResolveOptionalParams {
...
+  */
 export enum LuisApikeys {
     apiKeyHeader,
 }
 
+/**
+ * Luis Client Class
+ *
+ * @remarks
+ * This class is helper class for all the http request operations against the LUIS API
```
Comment (Maihav) [inline libraries/botbuilder-ai/src/luis-client/luisClient.ts]: I would say: `This is a helper class for...` or 'This class is a helper for...'

File: libraries/botbuilder-ai/src/luis-client/luisClient.ts [hunk 2]
```diff
@@ -51,11 +51,23 @@ export interface PredictionResolveOptionalParams {
...
+ * https://www.luis.ai/
+ */
 export class LuisClient {
+    /** Acceptable values: https://westus.api.cognitive.microsoft.com */
```
Comment (Maihav) [inline libraries/botbuilder-ai/src/luis-client/luisClient.ts]: What should that URL retrieve?

**Comment by Maihav** [comment]
Minor feedback for the `Specific Changes` section:

- It sounds better 'Add documentation to'

- `LuisApikeys` is missing the closing single quote

File: libraries/botbuilder-ai/src/luis-client/luisClient.ts (hunk 2 above)
Comment (GasparAcevedoZainSouthworks) [reply libraries/botbuilder-ai/src/luis-client/luisClient.ts]: @Konigstiger Can we replace this comment with something like [this](https://github.com/Azure/azure-sdk-for-node/blob/master/lib/services/luis/runtime/lib/lUISRuntimeClient.d.ts#L23)(This is the comment of the former implementation of the _LuisClient_)

File: libraries/botbuilder-ai/src/luis-client/luisClient.ts (hunk 1 above)
Comment (Konigstiger) [reply libraries/botbuilder-ai/src/luis-client/luisClient.ts]: Thank you, fixed.

File: libraries/botbuilder-ai/src/luis-client/luisClient.ts (hunk 2 above)
Comment (Konigstiger) [reply libraries/botbuilder-ai/src/luis-client/luisClient.ts]: It is just the allowed value of the _basePath variable. It is used internally by the constructor. We got feedback requesting that. We could only let the comment in the constructor. I will do that.

File: libraries/botbuilder-ai/src/luis-client/luisClient.ts [hunk 3]
```diff
@@ -51,12 +51,23 @@ export interface PredictionResolveOptionalParams {
...
+/**
+ * Luis Client Class
+ *
+ * @remarks
+ * This is a helper class for all the http request operations against the LUIS API.
+ * https://www.luis.ai/
+ */
 export class LuisClient {
-    private _basePath: string = '';
+    protected _basePath: string = '';
```
Comment (GasparAcevedoZainSouthworks) [inline libraries/botbuilder-ai/src/luis-client/luisClient.ts]: @Konigstiger I think that when you rebased against _master_ branch you reverted the change that made this member `private`. Also, I would suggest to leave the comment about what this member is, and what are its allowed values (something like [this](https://github.com/Azure/azure-sdk-for-node/blob/master/lib/services/luis/runtime/lib/lUISRuntimeClient.d.ts#L23) maybe?)

Comment (Konigstiger) [reply libraries/botbuilder-ai/src/luis-client/luisClient.ts]: I like that description, I will take your example. 
Also I will change the field to private too, as you said that was resolved before in other PR. Thanks!

Comment (Konigstiger) [reply libraries/botbuilder-ai/src/luis-client/luisClient.ts]: I like that description, I will take your example.
Also I will change the field to private too, as you said that was resolved before in other PR. Thanks!

**Pull request merged**